OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs

```diff
@@ -68,57 +84,113 @@ public void Tick(Squad owner)
 
 	class GroundUnitsAttackMoveState : GroundStateBase, IState
 	{
+		const int MaxAttemptsToAdvance = 6;
+		const int MakeWayTicks = 2;
+
+		// Give tolerance for AI grouping team at start
+		int attemptsToAdvance = MaxAttemptsToAdvance * 3;
+		int makeWay = MakeWayTicks;
+		WPos lastPos = WPos.Zero;
+
 		public void Activate(Squad owner) { }
 
 		public void Tick(Squad owner)
 		{
+			// Basic check
 			if (!owner.IsValid)
 				return;
 
 			if (!owner.IsTargetValid)
 			{
-				var closestEnemy = FindClosestEnemy(owner);
-				if (closestEnemy != null)
-					owner.TargetActor = closestEnemy;
+				var targetActor = GetRandomPreferredTarget(owner);
+				if (targetActor != null)
+					owner.TargetActor = targetActor;
 				else
 				{
 					owner.FuzzyStateMachine.ChangeState(owner, new GroundUnitsFleeState(), true);
 					return;
 				}
 			}
 
-			var leader = owner.Units.ClosestTo(owner.TargetActor.CenterPosition);
+			// Initialize leader. Optimaze pathfinding by using leader.
```

Comment:
This still has a typo (`Optimaze`) and doesn't make much sense. This comment should really instead be explaining why this is using `owner.Units.FirstOrDefault()` and not `owner.Units.ClosestTo(owner.TargetActor.CenterPosition);` which is still used in the `NavyUnitsAttackMoveState`.

Reply:
I plan to wait for this PR merged then to fix `NavyUnitsAttackMoveState`.

Reply:
typo fixed